contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -34,8 +41,19 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
     uint256 public maxWithdrawalDeviation;
     // Max deposit deviation denominated in 1e18 (1e18 == 100%)
     uint256 public maxDepositDeviation;
+    // Rate providers of the Balancer pool
+    IRateProvider[] public poolRateProvidersCache;
```

Comment:
You could check this, but I think a dynamic array reads the length storage slot first to bounds check, then the data slot.

That said, this is clean, and should be a lot less gas than what we were doing. Probably good to keep the clean way.